### PR TITLE
Addressing the Latest Kubernetes NGINX Ingress Controller CVE-2024-7646 Vulnerability

### PR DESCRIPTION
**Pull request by anuddeeph1**
## Description
This policy ensures that Ingress resources do not have certain disallowed annotations and that the ingress-nginx controller Pod is running an appropriate version of the image. It checks for the presence of the `nginx.ingress.kubernetes.io/server-snippet` annotation and disallows its usage, enforces specific values for `auth-tls-verify-client`, and ensures that the ingress-nginx controller image is of the required version.

## Checklist
- [] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.
- [] I have added the artifacthub-pkg.yml file and have verified it is complete and correct.

### CODE REVIEW

**Comment by realshuting** [comment]
Hi @anuddeeph1 - there are 11 failures, can you please take a look?

**Comment by anuddeeph1** [comment]
> Hi @anuddeeph1 - there are 11 failures, can you please take a look?

Hi @realshuting, The 11 failures is for different policies, The failures are not related to this policy

File: other/check-ingress-nginx-controller-version-and-annotation-policy/.chainsaw-test/chainsaw-test.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  - name: step-02
+    try:
+    - patch:
+        resource:
+          apiVersion: kyverno.io/v1
+          kind: ClusterPolicy
+          metadata:
+            name: check-ingress-nginx-controller-version-and-annotation-policy
+          spec:
+            validationFailureAction: Enforce
```
Comment (realshuting) [inline other/check-ingress-nginx-controller-version-and-annotation-policy/.chainsaw-test/chainsaw-test.yaml:21]: Why not set `Enforce` in the policy directly?

Comment (anuddeeph1) [reply other/check-ingress-nginx-controller-version-and-annotation-policy/.chainsaw-test/chainsaw-test.yaml:21]: Hi @realshuting, I've been using audit mode in the policy, but for testing purposes, I'm switching to enforce mode to ensure that non-compliant pods are blocked, and the test passes.

**Review by realshuting — APPROVED** [review]

**Pull request merged**